apps/src/blockly/addons/functionEditor.js

```diff
@@ -166,11 +170,24 @@ export default class FunctionEditor {
         deletable: false,
         movable: false,
       };
+
       this.block = Blockly.serialization.blocks.append(
         this.addEditorWorkspaceBlockConfig(newDefinitionBlock),
         this.editorWorkspace
       );
     }
+    const type = procedureType || existingProcedureBlock.type;
+    const isBehavior = type === BLOCK_TYPES.behaviorDefinition;
+    const existingFrame = this.editorWorkspace.svgFrame_;
```

Comment:
nit: move the check for and disposal of `existingFrame` to `clearEditorWorkspace`